src/ImageSharp/Metadata/Profiles/IPTC/IptcProfile.cs

```diff
@@ -263,17 +287,24 @@ public void UpdateData()
                 // |           | Octet Count    | the following data field(32767 or fewer octets). Note that the value of bit 7 of|
                 // |           |                | octet 4(most significant bit) always will be 0.                                 |
                 // +-----------+----------------+---------------------------------------------------------------------------------+
-                this.Data[i++] = IptcTagMarkerByte;
-                this.Data[i++] = 2;
-                this.Data[i++] = (byte)value.Tag;
-                this.Data[i++] = (byte)(value.Length >> 8);
-                this.Data[i++] = (byte)value.Length;
-                if (value.Length > 0)
-                {
-                    Buffer.BlockCopy(value.ToByteArray(), 0, this.Data, i, value.Length);
-                    i += value.Length;
-                }
+                offset = this.WriteRecord(offset, value.ToByteArray(), IptcRecordNumber.Application, (byte)value.Tag);
+            }
+        }
+
+        private int WriteRecord(int offset, byte[] recordData, IptcRecordNumber recordNumber, byte recordBinaryRepresentation)
+        {
+            this.Data[offset++] = IptcTagMarkerByte;
+            this.Data[offset++] = (byte)recordNumber;
+            this.Data[offset++] = recordBinaryRepresentation;
+            this.Data[offset++] = (byte)(recordData.Length >> 8);
+            this.Data[offset++] = (byte)recordData.Length;
```

Comment:
Here the JIT needs to introduce a bound check for each access to `Data`. These can be avoided by
```c#
Span<byte> data = this.Data.AsSpan(offset, 5);
data[0] = IptcTagMarkerByte;
data[1] = recordNumber;
data[2] = recordBinaryRepresentation;
data[3] = (byte)(recordData.Length >> 8);
data[4] = (byte)recordData.Length;
offset += 5;
```
So if the `AsSpan` succeeds the JIT knows that all index accesses are within bounds, so no bounds check needs to be emitted (tested with .NET 6).